master/buildbot/reporters/zulip.py

```diff
@@ -0,0 +1,58 @@
+from twisted.internet import defer
+
+from buildbot import config
+from buildbot.reporters import utils
+from buildbot.reporters.http import HttpStatusPushBase
+from buildbot.util import httpclientservice
+from buildbot.util.logger import Logger
+
+logger = Logger()
+
+
+class ZulipStatusPush(HttpStatusPushBase):
+    name = "ZulipStatusPush"
+
+    def checkConfig(self, endpoint, token, stream=None, **kwargs):
+        if not isinstance(endpoint, str):
+            config.error("Endpoint must be a string")
+        if not isinstance(token, str):
+            config.error("Token must be a string")
+
+    @defer.inlineCallbacks
+    def reconfigService(self, endpoint, token, stream=None, **kwargs):
+        super().reconfigService(**kwargs)
+        self._http = yield httpclientservice.HTTPClientService.getService(
+            self.master, endpoint,
+            debug=self.debug, verify=self.verify)
+        self.token = token
+        self.stream = stream
+
+    def buildStarted(self, key, build):
```

Comment:
I think you should be able to override `wantedDetails` class variable to get the needed data to be sent to `send()`. This way you wouldn't need to override `buildStarted` and `buildFinished` and extract build details yourself in `send()`.

Reply:
For an example, look at `GerritVerifyStatusPush`.